### ISSUE TITLE
Internal Consistency check failed

### ISSUE THREAD

**Issue by johnhamelink** [processone/xmpp#19]
Hi there,

I'm having an issue with this library on the latest version of OSX (10.13.1) and erlang 20.1:

```
Erlang/OTP 20 [erts-9.1] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:10] [hipe] [kernel-poll:false]

<snip>

===> Fetching rebar3_hex ({pkg,<<"rebar3_hex">>,<<"4.1.0">>})
===> Downloaded package, caching at /Users/john/.cache/rebar3/hex/default/packages/rebar3_hex-4.1.0.tar
===> Compiling rebar3_hex
===> Fetching pc ({pkg,<<"pc">>,<<"1.6.0">>})
===> Downloaded package, caching at /Users/john/.cache/rebar3/hex/default/packages/pc-1.6.0.tar
===> Compiling pc
===> Compiling xmpp
===> Compiling /Users/john/Public/fap/api/2.1/deps/xmpp/c_src/jid.c
===> Linking /Users/john/Public/fap/api/2.1/deps/xmpp/priv/lib/jid.so
src/xep0191.erl:6: Warning: export_all flag enabled - all functions will be exported

src/xep0153.erl:6: Warning: export_all flag enabled - all functions will be exported

src/xep0231.erl:6: Warning: export_all flag enabled - all functions will be exported

src/xep0033.erl:6: Warning: export_all flag enabled - all functions will be exported

src/xep0352.erl:6: Warning: export_all flag enabled - all functions will be exported

src/xep0030.erl:6: Warning: export_all flag enabled - all functions will be exported

src/xep0356.erl:6: Warning: export_all flag enabled - all functions will be exported

src/xep0357.erl:6: Warning: export_all flag enabled - all functions will be exported

src/xep0023.erl:6: Warning: export_all flag enabled - all functions will be exported

src/xep0221.erl:6: Warning: export_all flag enabled - all functions will be exported

src/xep0355.erl:6: Warning: export_all flag enabled - all functions will be exported

src/xep0369.erl:6: Warning: export_all flag enabled - all functions will be exported

src/xep0220.erl:6: Warning: export_all flag enabled - all functions will be exported

src/xep0022.erl:6: Warning: export_all flag enabled - all functions will be exported

src/xep0092.erl:6: Warning: export_all flag enabled - all functions will be exported

src/xep0045.erl:6: Warning: export_all flag enabled - all functions will be exported

src/rfc6121.erl:6: Warning: export_all flag enabled - all functions will be exported

src/xep0131.erl:6: Warning: export_all flag enabled - all functions will be exported

src/rfc6120.erl:6: Warning: export_all flag enabled - all functions will be exported

src/xep0050.erl:6: Warning: export_all flag enabled - all functions will be exported

src/xep0078.erl:6: Warning: export_all flag enabled - all functions will be exported

src/xep0085.erl:6: Warning: export_all flag enabled - all functions will be exported

src/xep0279.erl:6: Warning: export_all flag enabled - all functions will be exported

===> Compiling src/xmpp_codec.erl failed
xmpp_codec: function get_mod/2+2657:
  Internal consistency check failed - please report this bug.
  Instruction: {test,bs_test_tail2,{f,856},[{x,5},0]}
  Error:       {no_bsm_context,{x,5}}:


src/xmpp_codec.erl:6: Warning: export_all flag enabled - all functions will be exported
```

I have no idea what this means or how to go about debugging it!

**Comment by weiss** [comment]
This is a [known][1] and [fixed][2] Erlang/OTP bug.  Use 20.0 until 20.2 is released.

[1]: https://bugs.erlang.org/browse/ERL-493
[2]: https://github.com/erlang/otp/pull/1588

**Comment by johnhamelink** [comment]
FYI on High sierra Erlang versions <20.1 do not work because of the way OSX have packaged boringssl: https://github.com/elixir-lang/elixir/issues/6200

**Comment by weiss** [comment]
Using Erlang/OTP 20.1.1 or newer from Git would also do the trick.

**Comment by johnhamelink** [comment]
Thanks @weiss I will give that a go

**Comment by johnhamelink** [comment]
@weiss confirmed fixed on 20.2. Thanks!

**Comment by zinid** [comment]
Oh, it's finally released. Thanks for the info :)
